### PR TITLE
Manual setting of the bandwidth

### PR DESCRIPTION
**Pull request by lukashergt**
This PR adds `bw_method` as an optional kwarg for KDE plots, allowing to manually set the bandwidth for scipy's `gaussian_kde`.

This can be particularly helpful for extremely correlated parameter combination, where the auto-determined bandwidth often is too big, as the following example demonstrates:
```python
data = np.random.randn(10000, 2)
data[:, 1] = data[:, 0]**2
ns = NestedSamples(data=data, columns=['x', 'y'])

fig, axes = make_2d_axes(['x', 'y'], upper=False, diagonal=False)
fig, axes = ns.plot_2d(axes, ncompress=10000, lower_kwargs=dict(bw_method=None), label="bw_method=None (default)")
fig, axes = ns.plot_2d(axes, ncompress=10000, lower_kwargs=dict(bw_method=0.05), label="bw_method=0.05")
fig, axes = ns.plot_2d(axes, ncompress=200, types=dict(lower='scatter'), c='k', label="underlying, equal weight samples")
axes.iloc[0, 0].set_xlim(-2.5, 2.5)
axes.iloc[0, 0].set_ylim(0, 5)
axes.iloc[0, 0].legend()
```
image

The black dots show the (compressed) underlying, equally weighted sample points. The blue contours show the default KDE behaviour, where scipy uses a rule-of-thumb to determine a bandwidth. For the orange contours the bandwidth has been manually decreased.
  

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: anesthetic/plot.py [hunk 1]
```diff
@@ -516,6 +516,7 @@ def kde_plot_1d(ax, data, *args, **kwargs):
         dict(linewidth=['lw'], linestyle=['ls'], color=['c'],
              facecolor=['fc'], edgecolor=['ec']))
 
+    bw_method = kwargs.pop('bw_method', None)
```
[the remark this answers is not in the document]
Comment (lukashergt) [reply anesthetic/plot.py:519]: In principle these kwargs are all covered by:
```
    This functions as a wrapper around matplotlib.axes.Axes.plot, with a kernel
    density estimation computation provided by scipy.stats.gaussian_kde in
    between. All remaining keyword arguments are passed onwards.
```
So a check of `scipy.stats.gaussian_kde` allows finding out about the `bw_method` kwarg. 

Should we add to the docs regardless?

[the remark this answers is not in the document]
Comment (lukashergt) [reply anesthetic/plot.py:519]: This is somewhat related to #22 on automatically inheriting kwarg documentation from the original matplotlib or scipy functions...

[the remark this answers is not in the document]
Comment (williamjameshandley) [reply anesthetic/plot.py:519]: No, I think you're right that this is covered.

**Review by williamjameshandley — APPROVED** [review]
Many thanks @lukashergt. Please squash and merge when ready.

**Pull request merged**